astropy/coordinates/representation.py

```diff
@@ -348,11 +376,15 @@ def __repr__(self):
         prefixstr = '    '
         arrstr = _array2string(self._values, prefix=prefixstr)
 
+        diffstr = ''
+        if getattr(self, 'differentials', None):
+            diffstr = '\n (has differentials: {0})'.format(
+                ', '.join([repr(key) for key in self.differentials.keys()]))
```

Comment:
I find the description here confusing because it implies the differentials *have that unit*.  How about "has differentials: d/d[s]" or something like that to make it clear its the *derivative* unit?    I think this might do it:
```
', '.join(['d/d[{}]'.format(key) for key in ks])
```